pcsx2/DEV9/DEV9.h

```diff
@@ -0,0 +1,669 @@
+#define SPD_R_UDMA_MODE (SPD_REGBASE + 0x74)
+
+
+/*
+ * SMAP (PS2 Network Adapter) register definitions.
+ *
+ * Copyright (c) 2003 Marcus R. Brown <mrbrown@0xd6.org>
+ *
+ * * code included from the ps2smap iop driver, modified by linuzappz  *
+ */
+
+
+/* SMAP interrupt status bits (selected from the SPEED device).  */
+#define SMAP_INTR_EMAC3 (1 << 6)
+#define SMAP_INTR_RXEND (1 << 5)
+#define SMAP_INTR_TXEND (1 << 4)
+#define SMAP_INTR_RXDNV (1 << 3) /* descriptor not valid */
+#define SMAP_INTR_TXDNV (1 << 2) /* descriptor not valid */
+#define SMAP_INTR_CLR_ALL (SMAP_INTR_RXEND | SMAP_INTR_TXEND | SMAP_INTR_RXDNV)
+#define SMAP_INTR_ENA_ALL (SMAP_INTR_EMAC3 | SMAP_INTR_CLR_ALL)
+#define SMAP_INTR_BITMSK 0x7C
+
+/* SMAP Register Definitions.  */
+
+#define SMAP_REGBASE (SPD_REGBASE + 0x100)
+
+#define SMAP_R_BD_MODE (SMAP_REGBASE + 0x02)
+#define SMAP_BD_SWAP (1 << 0)
+
+#define SMAP_R_INTR_CLR (SMAP_REGBASE + 0x28)
+
+/* SMAP FIFO Registers.  */
+
+#define SMAP_R_TXFIFO_CTRL (SMAP_REGBASE + 0xf00)
+#define SMAP_TXFIFO_RESET (1 << 0)
+#define SMAP_TXFIFO_DMAEN (1 << 1)
+#define SMAP_R_TXFIFO_WR_PTR (SMAP_REGBASE + 0xf04)
+#define SMAP_R_TXFIFO_SIZE (SMAP_REGBASE + 0xf08)
+#define SMAP_R_TXFIFO_FRAME_CNT (SMAP_REGBASE + 0xf0C)
+#define SMAP_R_TXFIFO_FRAME_INC (SMAP_REGBASE + 0xf10)
+#define SMAP_R_TXFIFO_DATA (SMAP_REGBASE + 0x1000)
+
+#define SMAP_R_RXFIFO_CTRL (SMAP_REGBASE + 0xf30)
+#define SMAP_RXFIFO_RESET (1 << 0)
+#define SMAP_RXFIFO_DMAEN (1 << 1)
+#define SMAP_R_RXFIFO_RD_PTR (SMAP_REGBASE + 0xf34)
+#define SMAP_R_RXFIFO_SIZE (SMAP_REGBASE + 0xf38)
+#define SMAP_R_RXFIFO_FRAME_CNT (SMAP_REGBASE + 0xf3C)
+#define SMAP_R_RXFIFO_FRAME_DEC (SMAP_REGBASE + 0xf40)
+#define SMAP_R_RXFIFO_DATA (SMAP_REGBASE + 0x1100)
+
+#define SMAP_R_FIFO_ADDR (SMAP_REGBASE + 0x1200)
+#define SMAP_FIFO_CMD_READ (1 << 1)
+#define SMAP_FIFO_DATA_SWAP (1 << 0)
+#define SMAP_R_FIFO_DATA (SMAP_REGBASE + 0x1208)
+
+/* EMAC3 Registers.  */
+
+#define SMAP_EMAC3_REGBASE (SMAP_REGBASE + 0x1f00)
+
+#define SMAP_R_EMAC3_MODE0_L (SMAP_EMAC3_REGBASE + 0x00)
+#define SMAP_E3_RXMAC_IDLE (1 << (15 + 16))
+#define SMAP_E3_TXMAC_IDLE (1 << (14 + 16))
+#define SMAP_E3_SOFT_RESET (1 << (13 + 16))
+#define SMAP_E3_TXMAC_ENABLE (1 << (12 + 16))
+#define SMAP_E3_RXMAC_ENABLE (1 << (11 + 16))
+#define SMAP_E3_WAKEUP_ENABLE (1 << (10 + 16))
+#define SMAP_R_EMAC3_MODE0_H (SMAP_EMAC3_REGBASE + 0x02)
+
+#define SMAP_R_EMAC3_MODE1 (SMAP_EMAC3_REGBASE + 0x04)
+#define SMAP_R_EMAC3_MODE1_L (SMAP_EMAC3_REGBASE + 0x04)
+#define SMAP_R_EMAC3_MODE1_H (SMAP_EMAC3_REGBASE + 0x06)
+#define SMAP_E3_FDX_ENABLE (1 << 31)
+#define SMAP_E3_INLPBK_ENABLE (1 << 30) /* internal loop back */
+#define SMAP_E3_VLAN_ENABLE (1 << 29)
+#define SMAP_E3_FLOWCTRL_ENABLE (1 << 28) /* integrated flow ctrl(pause frame) */
+#define SMAP_E3_ALLOW_PF (1 << 27)        /* allow pause frame */
+#define SMAP_E3_ALLOW_EXTMNGIF (1 << 25)  /* allow external management IF */
+#define SMAP_E3_IGNORE_SQE (1 << 24)
```

Comment:
I think we should keep the old formatting here, the tree-like format is important